src/script/script.h

```diff
@@ -498,6 +498,11 @@ class CScript : public CScriptBase
         return *this;
     }
 
+    CScript& operator<<(const std::span<const std::byte> b) LIFETIMEBOUND
+    {
+        return *this << BytesToUInt8s(b);
```

Comment:
Yes, that's what I tried at first, but I don't yet see how that would be possible without changing `prevector`, as explained [here](https://github.com/bitcoin/bitcoin/pull/30765#discussion_r1747621990)

Reply:
> Yes, that's what I tried at first, but I don't yet see how that would be possible without changing `prevector`, as explained [here](https://github.com/bitcoin/bitcoin/pull/30765#discussion_r1747621990)

Thanks that's helpful to know, and it's clear why changing prevector could fix this but not clear why it would be necessary to change prevector to fix that as opposed to just passing a start and end pointer of the type it is expecting

Reply:
Because we were casting back and forth this way. But I think I found a good enough generalization with your help, we don't need to connect the two operators this way.